File: Array/3.py
n = int(input())
num = 1
arr = [[0 for _ in range(n)] for _ in range(n)]
for i in range(n):
    if i % 2 == 0:
        for j in range(n):
            arr[i][j] = num
            num+=1
    else:
        for j in range(n-1, -1, -1):
            arr[i][j] = num
            num+=1
for i in range(n):
    for j in range(n):
        print(arr[i][j], end = " ")
    print()
            

File: Array/3.c
#include <stdio.h>

int main()
{
    int N;
    scanf("%d", &N);
    int arr[N][N];
    int num = 1;
    for (int i = 0; i < N; i++){
        if (i %2==0){
            for (int j = 0; j < N;j++){
                arr[i][j] = num;
                num++;
            }
        }
        else{
            for(int j = N-1; j >= 0; j--){
                arr[i][j] = num;
                num++;
            }
        }
    }
    for (int i = 0; i < N; i++){
        for (int j = 0; j < N; j++){
            printf("%d ", arr[i][j]);
        }
        printf("\n");
    }
}